### ISSUE TITLE
Пожалуйста добавьте поддержку новой версии

### ISSUE THREAD

**Issue by delef** [mblsha/yandex-music-mac-player#33]
Новая версия (редизайн) намного функциональнее и удобнее, возвращаюсь на старую и разочаровываюсь...

P.S.
Насколько понял по-именам - русскоязычные, так что не стал писать на английском.

**Comment by shamrin** [comment]
@delef Спасибо за интерес! Я перестал пользоваться этой программой и перешёл на Chrome+Keysocket. Плюсы полноценного браузера перевесили проблемы с реализацией media-клавиш в Chrome. Поэтому личного интереса патчить yandex-music-mac-player у меня уже нет.

Тем не менее, если кто-нибудь напишет патч с поддержкой новой Я.Музыки, я его с удовольствием включу :-)

**Comment by delef** [comment]
Очень жаль, что я не пишу на Objective-C(

Спасибо за совет) Тогда пока тоже на Keysocket перескочу, может появится свободное время и на Ruby напишу, он вроде тоже неплох для создания приложений под Mac.. homebrew - показатель, хоть и консольный.

В общем, спасибо)

**Comment by shamrin** [comment]
@delef Я переписал эту программу, совсем не зная Objective-C :-) Оказалось, не так уж трудно научиться. XCode и Stackoverflow помогают в сложных случаях.

Скорее всего, менять много не надо и только кусочки на JavaScript. См. вызовы функции `eval` в `YandexMusicAppDelegate.m`: [1][], [2][], [3][]. 

[1]: https://github.com/mblsha/yandex-music-mac-player/blob/dba5e6cc1e960418f9188f87251693204ea28396/YandexMusic/YandexMusicAppDelegate.m#L129-L138
[2]: https://github.com/mblsha/yandex-music-mac-player/blob/dba5e6cc1e960418f9188f87251693204ea28396/YandexMusic/YandexMusicAppDelegate.m#L168-L169
[3]: https://github.com/mblsha/yandex-music-mac-player/blob/dba5e6cc1e960418f9188f87251693204ea28396/YandexMusic/YandexMusicAppDelegate.m#L181-L197

Ещё может помочь изучение [реализации Я.Музыки в KeySocket][ks].

[ks]: https://github.com/borismus/keysocket/blob/master/extension/keysocket-yandex-music.js

**Comment by shamrin** [comment]
@delef По поводу переписывания на Ruby: насколько я помню, гораздо больше времени я потратил на возню с OS X API, а не на Objective C. Не думаю, что Ruby как-то облегчит дело (возможно, даже наоборот, потому что придётся какие-то обёртки использовать).

**Comment by delef** [comment]
Согласен.. значит буду вникать) Есть вариант реализации приложения (полноценного) на основе "несуществующего" api яндекс.музыки) Реализация уже есть)

И.. я думаю работать быстрее будет и интерфейс под Yosemite подстроить можно, но это только задумки, которые я надеюсь реализую. Просто становится вопрос - а нужен ли вообще он, если есть KeySocket?

Попользовался им час и понял, что всё неплохо у них.

**Comment by shamrin** [comment]
Да, KeySocket в целом хорош, за исключением нескольких проблем:

1. Необходимость ковыряться в настройках Хрома после установке расширения.
2. Конфликты с VLC и другими программами, которые ловят media-клавиши. Некоторые подробности есть в https://github.com/borismus/keysocket/issues/78. Это проблема Хрома.

**Comment by delef** [comment]
Понял, учту) Спасибо)

**Comment by msoap** [comment]
Для себя я поправил под новую я.музыку, но только шорткаты prev/next/play/stop. Сходу не получилось достать имя трека для показа в меню. Прислать pullrequest? :)

**Comment by delef** [comment]
Пришли, это же уже хорошо)

А в чем проблема с именем трека? Там же только контент тега забрать..

**Comment by delef** [comment]
```
<div class="track__name-wrap">
    <a href="/album/625175/track/3591615" class="track__title link" title="Born To Die">Born To Die</a>
    <div class="track__artists"><a href="/artist/643850" class="link link_mute">Lana Del Rey</a></div>
</div>
```

**Comment by msoap** [comment]
Там хитрее, внедряется js код, который вешается на событие смены трека, чтобы сама страница оповещала приложение о смене трека и о статусе (пауза/играет).

Вот это место в YandexMusic/YandexMusicAppDelegate.m:
```Objective-C
    // inject JavaScript handler to track playing/non-playing state
    [self eval:@"(function() { "
                   "if(window.MAC_APP_HANDLER_INSTALLED) { yamusicapp.log(\"handler already there\"); return; } "
                   "yamusicapp.log(\"installing handler\"); "
                   "yamusicapp.notify(false); "
                   "Mu.events.bind(\"pl:onPlayTrack pl:onPause pl:onResume pl:onStop\", function(f) { "
                     "var isPlaying = (f.type === \"pl:onResume\" || f.type === \"pl:onPlayTrack\"); "
                     "yamusicapp.notify(isPlaying); "
                   "}); "
                   "window.MAC_APP_HANDLER_INSTALLED = true; "
                "})();"];
```

Поэтому, кстати, у меня еще не меняется иконка и название в меню (Play/Pause)

**Comment by msoap** [comment]
Без учета статусов/имени трека - https://github.com/mblsha/yandex-music-mac-player/pull/34
